shell/components/fleet/dashboard/ResourcePanel.vue

```diff
@@ -0,0 +1,357 @@
+
+      const canvas = document.createElement('canvas');
+
+      canvas.id = this.chartId;
+      canvas.className = 'chart';
+
+      container.append(canvas);
+
+      const data = this.buildChartData();
+      const options = this.chartOptions;
+
+      const chart = new Chart(canvas, {
+        type: 'doughnut',
+        data,
+        options,
+      });
+
+      this.chart = markRaw(chart);
+    }
+  },
+
+  computed: {
+    ...mapGetters({ theme: 'prefs/theme' }),
+
+    statePanels() {
+      const out = [];
+
+      this.states.forEach((state) => {
+        const {
+          statePanel,
+          resources
+        } = state;
+
+        const exists = out.find((s) => s.id === statePanel.id);
+
+        if (exists) {
+          exists.count += resources.length;
+        } else {
+          out.push({
+            ...statePanel,
+            state,
+            count: resources.length
+          });
+        }
+      });
+
+      return out;
+    },
+
+    count() {
+      return this.statePanels.reduce((acc, state) => acc + state.count, 0);
+    },
+
+    typeLabel() {
+      return this.t(`typeLabel."${ this.type }"`, { count: this.count });
+    },
+  },
+
+  watch: {
+    statePanels: {
+      handler: 'updateStates',
+      deep:    true,
+    },
+    theme() {
+      if (this.chart) {
+        this.chart.update();
+      }
+    },
+  },
+
+  methods: {
+    buildChartData() {
+      const chartStates = FleetUtils.dashboardStates;
+
+      const labels = chartStates.map(({ label }) => label);
+      const backgroundColor = chartStates.map(({ color }) => color);
+
+      const data = this.getChartStates(this.statePanels);
+
+      return {
+        labels,
+        datasets: [
+          {
+            data,
+            backgroundColor,
+          },
+        ],
+      };
+    },
+
+    updateStates(value) {
+      if (this.chart) {
+        this.chart.data.datasets.forEach((dataset) => {
+          dataset.data = this.getChartStates(value);
+        });
+
+        this.chart.update();
+      }
+    },
+
+    getChartStates(states) {
+      return FleetUtils.dashboardStates.map(({ id }) => states.find((s) => s.id === id)?.count || 0);
+    },
+
+    selectState(state) {
+      if (!this.selectable) {
+        return;
+      }
+
+      this.$emit('click:state', state.id);
+    },
+
+    onClickBadge(state) {
+      this.selectState(state);
+    },
+
+    bgColor() {
+      return getComputedStyle(document.body).getPropertyValue('--body-bg');
+    }
+  }
+};
+</script>
+
+<template>
+  <div class="panel">
+    <div
+      v-if="showChart"
+      :id="chartId + '-container'"
+      :data-testid="'chart-container'"
+      class="panel-chart-container"
+    />
+    <div class="details">
+      <div
+        class="description"
+        :data-testid="'description'"
+      >
+        <span class="count">{{ count }}</span>
+        <span class="label">{{ typeLabel }}</span>
+      </div>
+      <div class="states">
+        <BadgeState
```

Comment:
The focus ring for the badges has a white border in dark mode

![image](https://github.com/user-attachments/assets/b0581c07-0404-473b-975a-6d7d2a2d3c8d)

Reply:
We need to involve UX for this, for now I'm changing to default `primary` color